ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -1033,10 +1037,13 @@ let list_required_vdis ~__context ~self =
 
 (* Find the SRs of all VDIs which have VBDs attached to the VM. *)
 let list_required_SRs ~__context ~self =
-  let vdis = list_required_vdis ~__context ~self in
-  let srs = List.map (fun vdi -> Db.VDI.get_SR ~__context ~self:vdi) vdis in
-  let srs = List.filter (fun sr -> Db.SR.get_content_type ~__context ~self:sr <> "iso") srs in
-  List.setify srs
+  list_required_vdis ~__context ~self
+  |> List.to_seq
+  |> Seq.map (fun self -> Db.VDI.get_SR ~__context ~self)
+  |> Seq.filter (fun self -> Db.SR.get_content_type ~__context ~self <> "iso")
```

Comment:
changing to seq here (and above) might affect the order in which DB calls are done, but since we aren't doing any writing we should be fine :+1:

Reply:
If there's a race condition with another operation, interleaving the `get_SR` and `get_content_type` calls might change the end result, but the issue would be more pronounced before, as not interleaving the calls separates them further in time.

Reply:
i.e. this is actually better :+1: